book: indicate that at least one error marker is required (#14589)
rust-lang/rust-clippy: book/src/development/adding_lints.md

```

Note that we are adding comment annotations with the name of our lint to mark

> lines where we expect an error. Except for very specific situations, at least

Reviewer: ```suggestion
lines where we expect an error. Except for very specific situations (`//@check-pass`), at least
```

Or just mention the thing which allows that in parentheses. Then someone knows how to mark them and also what that means, if they encounter it.

```

Note that we are adding comment annotations with the name of our lint to mark

> lines where we expect an error. With the exception of very specific situations
> identified with a `//@check-pass` annotation, at least one error marker must be
> present in a test file for it to be accepted.

Reviewer: ```suggestion
lines where we expect an error. Except for very specific situations, at least one error marker must be
present in a test file for it to be accepted. Exceptional test can be identified with a `//@check-pass` annotation.
```

Suggestion messes up formatting, so better copy and paste it.

I would swap the wording: Mention the common case first and then describe what to do in the exceptional case.

Author: I would even remove the `//@check-pass` mention, this is too specialized for this section, and only apply when you have, for example, `ui-toml` tests with specific configurations.

Reviewer: I think also for a bunch of `crashes/ice-1234.rs` tests, I think.

But yeah, not really necessary.
